streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -789,7 +789,10 @@ private int putNodeGroupName(final String nodeName,
         return newNodeGroupId;
     }
 
-    public synchronized ProcessorTopology build() {
+    /**
+     * @return the full topology minus any global state
+     */
+    public synchronized ProcessorTopology buildTopology() {
```

Comment:
This is just a side thing that's been bugging me. There actual changes contained in this PR are pretty small, and I'll point them all out specifically in the comments. But I can remove these method renamings if they create too much noise in the diff to review